search/src/main/java/au/com/d2dcrc/ia/search/management/EpgIngestionTask.java

```
@@ -63,10 +63,11 @@ public void run() {
             // Update the meta index status.
             setIndexStatus(EpgStatus.COMPLETE);
             logger.debug("Successfully ingested EPG data into index: " + indexName);
-        } catch (RuntimeException e) {
+        } catch (Throwable e) {
             logger.error("Failed to properly ingest EPG data into index: " + indexName, e);
             // Signal the failure in the meta index status.
             setIndexStatus(EpgStatus.ERROR);
```

I suspect when I tried to ingest my large dataset, OOM was thrown and skipped this, thus killing the thread. Nothing was collecting the future of the submittable, so the exception was never propagated to the main thread, thus the app still run with the status stuck in INDEXING state

---

It was entirely expected OOM would be thrown in the current way of doing things, but we should still get some logging, if we trigger it accidentailly